lib/browser-auto-tracking.js

```diff
@@ -87,6 +87,122 @@ export default function initAutoTrackingCore(lib) {
+          y=l.getElementsByTagName(r)[0];
+          y.parentNode.insertBefore(t,y);
+      })(window, document, "clarity", "script", microsoft_project_id);
+
+      //todo (clarity): how should we actually get/set cookies? (this is taken from clarity)
+      function getCookie(key) {
+        let cookies = document.cookie.split(';');
+        if (cookies) {
+          for (let i = 0; i < cookies.length; i++) {
+            let pair = cookies[i].split('=');
+            if (pair.length > 1 && pair[0] && pair[0].trim() === key) {
+              return pair[1];
+            }
+          }
+        }
+        return null;
+      }
+
+      let rootDomain = null;
+      function setCookie(key, value, time){
+        let expiry = new Date();
+        expiry.setDate(expiry.getDate() + time);
+        let expires = expiry ? "expires=" + expiry.toUTCString() : "";
+        let cookie = `${key}=${value};${expires};path=/`;
+        try {
+          // Attempt to get the root domain only once and fall back to writing cookie on the current domain.
+          if (rootDomain === null) {
+            let hostname = location.hostname ? location.hostname.split('.') : [];
+            // Walk backwards on a domain and attempt to set a cookie, until successful
+            for (let i = hostname.length - 1; i >= 0; i--) {
+              rootDomain = `.${hostname[i]}${rootDomain ? rootDomain : ""}`;
+              // We do not wish to attempt writing a cookie on the absolute last part of the domain, e.g. .com or .net.
+              // So we start attempting after second-last part, e.g. .domain.com (PASS) or .co.uk (FAIL)
+              if (i < hostname.length - 1) {
+                // Write the cookie on the current computed top level domain
+                document.cookie = `${cookie};domain=${rootDomain}`;
+                if (getCookie(key) === value) { return; }
+              }
+            }
+            // Finally, if we were not successful and gone through all the options, play it safe and reset rootDomain to be empty
+            // This forces our code to fall back to always writing cookie to the current domain
+            rootDomain = "";
+          }
+        } catch { rootDomain = ""; }
+        document.cookie = rootDomain ? `${cookie};domain=${rootDomain}` : cookie;
+      }
+
+      //todo (clarity): figure out if you send messages before you set the correct session (first verify this actually works across sessions)
+      const setClskCookie = function() {
+        // this is a hack to manipulate the sessionId sent with the clarity events to make up for the "identify" call weirdness
+        try {
+          let sessionUuid = session_cookie.get('session_uuid');
+          let clskCookie = getCookie('_clsk');
+
+          if (sessionUuid && clskCookie) {
+            const clskDetails = clskCookie.split('|');
+
+            const oldSessionUuid = clskDetails[0];
+            if (oldSessionUuid.length > 31 && oldSessionUuid !== sessionUuid) {
```

Comment:
Why 31?

Reply:
Good catch.  A UUID is 32 characters (before formatting characters).  I wanted to make sure there was an actual UUID in there, rather than Clarity's default session name (a short random string).